src/model/content.rs

```diff
@@ -346,6 +346,36 @@ impl Content {
     pub fn set_location(&mut self, location: Location) {
         self.attrs.push(Attr::Location(location));
     }
+
+    /// Gives an iterator over the children of this content
+    pub fn children(&self) -> impl Iterator<Item = &Content> {
+        self.attrs.iter().filter_map(Attr::child)
```

Comment:
Note that only styled and sequence nodes will have `Attr::Child` attributes. Others store their children in named fields. E.g. `table.children` is `Field("children", Array)`, so your code won't find that. If you want to find these too, you would need to look for fields of type `Content`. I tried to make a design with uniform children, but there are many different cases: Zero children, one, many, 2d children (matrix), named children (math.attach), etc. So that went nowhere. The current `Attr` representation is a bit weird because it shoves the possible fields of Styled, Sequence, and all others together in one enum. I intend to rewrite this.

Reply:
Yeah, I have already ran into issue with this in my testing, I will try and figure out if I can do something temporary here. I do think that a proper search function for subtrees (with early evaluation?) would be very powerful and make these kinds of features much easier in the future